book/visual-effects.md

```diff
@@ -829,148 +875,111 @@ class BlockLayout:
 
-Blending and stacking
-=====================
 
-Drawing shapes quickly is already a challenge, but with multiple
-shapes there's an additional question: what color should the pixel be
-when two shapes overlap? So far, our browser has only handled opaque
-shapes,[^nor-subpixel] and the answer has been simple: take the color
-of the top shape. But now we need more nuance.
 
-[^nor-subpixel]: It also hasn't considered subpixel geometry or
-    anti-aliasing, which also rely on color mixing.
+Blending and isolation
+======================
 
 Many objects in nature are partially transparent: frosted glass,
-clouds, or colored paper, for example. Looking through one, you see
+clouds, or tissue paper, for example. Looking through one, you see
 multiple colors *blended* together. That's also why computer screens work:
 the red, green, and blue lights [blend together][mixing] and appear to
-our eyes as another color. Designers use this effect[^mostly-models]
+our eyes as another color.[^subpixel] Designers use this effect[^mostly-models]
 in overlays, shadows, and tooltips, so our browser needs to support
 color mixing.
 
+[^subpixel]: Another subtle place blending matters is subpixel geometry and
+    anti-aliasing, which also rely on color mixing.
+
 [mixing]: https://en.wikipedia.org/wiki/Color_mixing
 
 [^mostly-models]: Mostly. Some more advanced blending modes on the web are
 difficult, or perhaps impossible, in real-world physics.
 
-Color mixing means we need to think carefully about the order of
-operations. For example, consider black text on an orange background,
-placed semi-transparently over a white background. The text
-is gray while the background is yellow-orange. That's due to blending:
-the text and the background are both partially transparent and let
-through some of the underlying white:
-
-<div style="opacity: 0.5; background: orange; color: black; font-size: 50px;
-    padding: 15px; text-align: center;flex:1;">Text</div>
-
-But importantly, the text isn't orange-gray: even though the text is
-partially transparent, none of the orange shines through. That's
-because the order matters: the text is *first* blended with the
-background; since the text is opaque, its blended pixels are black and
-overwrite the orange background. Only *then* is this black-and-orange
-image blended with the white background. Doing the operations in a
-different order would lead to dark-orange or black text.
-
-To handle this properly, browsers apply blending not to individual shapes but to
-a tree of surfaces. Conceptually, each surface is drawn individually,
-and then blended into its parent surface. Rastering a web page requires a
-bottom-up traversal of the tree: to raster a surface you first need to raster
-its contents, including its child stacking contexts, and then the whole
-contents need to be blended together into the parent.[^stacking-context-disc]
+Unfortunately, blending can depend on the order in which you mix the
```

Comment:
Can you tell me again the rationale for rewriting this whole section? It seems to be about the same subject, just with different words and examples. Was the old text super confusing?

Reply:
My guess is we can fix bugs & merge other stuff and rewrite a lot less. I started by rewriting & found lots of stuff to fix. I’ll make stand-alone PRs and see where we land.